var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -216,6 +216,13 @@ def build_targets(self):
 
         return targets
 
+    # QMCPACK prefers taking MPI compiler wrappers as CMake compilers.
+    def setup_build_environment(self, env):
+        spec = self.spec
+        if '+mpi' in spec:
+            env.set('CC', spec['mpi'].mpicc)
+            env.set('CXX', spec['mpi'].mpicxx)
```

Comment:
CMake also may have a `CMAKE_C_COMPILER` variable which might be "safer" but this is fine as-is.

Reply:
When `~mpi`, I guess environment variables `CC`, `CXX` are how spack pass compilers to CMake. So just use the same route. On the CMake side, it find the same compiler in both ways.